react/utils/client/links/uriSwitchLink.ts

```diff
@@ -36,17 +18,21 @@ interface OperationContext {
   runtime: RenderRuntime,
 }
 
-export const createUriSwitchLink = (baseURI: string, workspace: string) =>
+const hashFromExtensions = ext => ext && ext.persistedQuery && ext.persistedQuery.sha256Hash
+
+export const createUriSwitchLink = (baseURI: string, runtime: RenderRuntime) =>
   new ApolloLink((operation: Operation, forward?: NextLink) => {
-    const {query} = operation
-    const assets = assetsFromQuery(operation.query)
+    const {workspace, cacheHints} = runtime
```

Comment:
We should get `cacheHints` from context, just like `appsEtag` below. It's safer because if the runtime data gets updated with `fetchRuntime`, we will have the correct `cacheHints` here.

You must add `cacheHints` to `createRuntimeContextLink` in `RenderProvider` for that.